weaver/sdks/fabric/interoperation-node-sdk/src/SatpAssetManager.ts

```diff
@@ -0,0 +1,1111 @@
+        const waitForClaim = (contract, contractId, assetType, assetId, recipientECert, lockerECert, hashPreimage) => {
+            resolve(hashPreimage);
+        };
+        StartHTLCAssetClaimListener(contract, contractId, assetType, assetId, recipientECert, lockerECert, waitForClaim);
+    });
+}
+
+const HTLCAssetUnlocked = async (
+    contract: Contract,
+    contractId: string,
+    assetType: string,
+    assetId: string,
+    recipientECert: string,
+    lockerECert: string,
+): Promise<void> => {
+    return new Promise((resolve, reject) => {
+        const waitForUnlock = (contract, contractId, assetType, assetId, recipientECert, lockerECert) => {
+            resolve();
+        };
+        StartHTLCAssetUnlockListener(contract, contractId, assetType, assetId, recipientECert, lockerECert, waitForUnlock);
+    });
+}
+
+const HTLCFungibleAssetLocked = async (
+    contract: Contract,
+    contractId: string,
+    assetType: string,
+    numUnits: number,
+    recipientECert: string,
+    lockerECert: string,
+): Promise<string> => {
+    return new Promise((resolve, reject) => {
+        const waitForLock = (contract, contractId, assetType, numUnits, recipientECert, lockerECert, hashValue) => {
+            resolve(hashValue);
+        };
+        StartHTLCFungibleAssetLockListener(contract, contractId, assetType, numUnits, recipientECert, lockerECert, waitForLock);
+    });
+}
+
+const HTLCFungibleAssetClaimed = async (
+    contract: Contract,
+    contractId: string,
+    assetType: string,
+    numUnits: number,
+    recipientECert: string,
+    lockerECert: string,
+): Promise<string> => {
+    return new Promise((resolve, reject) => {
+        const waitForClaim = (contract, contractId, assetType, numUnits, recipientECert, lockerECert, hashPreimage) => {
+            resolve(hashPreimage);
+        };
+        StartHTLCFungibleAssetClaimListener(contract, contractId, assetType, numUnits, recipientECert, lockerECert, waitForClaim);
+    });
+}
+
+const HTLCFungibleAssetUnlocked = async (
+    contract: Contract,
+    contractId: string,
+    assetType: string,
+    numUnits: number,
+    recipientECert: string,
+    lockerECert: string,
+): Promise<void> => {
+    return new Promise((resolve, reject) => {
+        const waitForUnlock = (contract, contractId, assetType, numUnits, recipientECert, lockerECert) => {
+            resolve();
+        };
+        StartHTLCFungibleAssetUnlockListener(contract, contractId, assetType, numUnits, recipientECert, lockerECert, waitForUnlock);
+    });
+}
+
+export {
+    createAssetExchangeAgreementSerialized,
+    createFungibleAssetExchangeAgreementSerialized,
+    createAssetLockInfoSerialized,
+    createAssetClaimInfoSerialized,
+    createHTLC,
+    createFungibleHTLC,
+    claimAssetInHTLC,
+    claimAssetInHTLCusingContractId,
+    claimFungibleAssetInHTLC,
+    reclaimAssetInHTLC,
+    reclaimAssetInHTLCusingContractId,
+    reclaimFungibleAssetInHTLC,
+    assignAsset,
```

Comment:
Just to confirm this file is duplicate of `AssetManager.ts` right? except this new function, can you remove all other duplicate code in this file?

Reply:
Done